src/main/java/meeju/Storage.java

```diff
@@ -0,0 +1,104 @@
+package meeju;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * This class handles the reading and writing of tasks to and from file.
+ */
+public class Storage {
+    private File file;
+    private Scanner scanner;
+    private String path = "./data/meeju.txt";
+
+    /* Note - The delimiter used is '!-' */
+
+    /**
+     * Constructor for Storage
+     * If the directory for the storage file does not exist, it creates the data directory.
+     */
+    public Storage() {
+        this.file = new File(this.path);
+        File directory = file.getParentFile();
+        if (!directory.exists()) {
+            directory.mkdirs();
+        }
+    }
+
+    /**
+     * Initializes the task list by reading from the storage file.
+     *
+     * This method is invoked at the start of the program when TaskList is initialised.
+     * It reads from the file and populates the tasklist with stored data
+     *
+     * @return An ArrayList containing the tasks initialized from the file.
+     */
+    public ArrayList<Task> initialiseList() {
+        ArrayList<Task> taskList = new ArrayList<>();
+        if (file.exists()) {
+            try {
+                this.scanner = new Scanner(this.file);
+            } catch (Exception e) {
+                System.out.println(e); //Ideally never reached here!
+            }
+            while (scanner.hasNext()) {
+                String currentLine = scanner.nextLine();
+                String[] currentLineParsed = currentLine.split("!-");
```

Comment:
Should the variable name be `currentParsedLines` to follow the plural convention?